tests/js/server/aql/aql-optimizer-indexes.js

```diff
@@ -1577,438 +1391,772 @@ function optimizerIndexesTestSuite () {
+        let q = "FOR doc IN UnitTestsCollection FILTER doc.value == 2 && doc.test1 == 'abc' && (doc.test2 || doc.test3) RETURN doc";
+        var results = AQL_EXECUTE(q);
+        assertEqual(1, results.json.length);
+        assertEqual(2, results.json[0].value);
+        assertEqual(v[0], results.json[0].test1);
+        assertEqual(v[1], results.json[0].test2);
+        assertEqual(v[2], results.json[0].test3);
 
+        let plan = AQL_EXPLAIN(q).plan;
+        let nodes = plan.nodes;
+        let nodeTypes = plan.nodes.map(function(node) {
+          return node.type;
+        });
+        let indexNode = nodes[nodeTypes.indexOf("IndexNode")];
+        assertEqual("n-ary or", indexNode.condition.type);
+        assertEqual(2, indexNode.condition.subNodes.length);
+        assertEqual("n-ary and", indexNode.condition.subNodes[0].type);
+        assertEqual("compare ==", indexNode.condition.subNodes[0].subNodes[0].type);
+        assertEqual("n-ary and", indexNode.condition.subNodes[1].type);
+        assertEqual("compare ==", indexNode.condition.subNodes[1].subNodes[0].type);
+      });
+    },
 ////////////////////////////////////////////////////////////////////////////////
 /// @brief test index usage
 ////////////////////////////////////////////////////////////////////////////////
 
-    testIndexOrSkiplistMultipleRanges : function () {
-      c.ensureSkiplist("value2", "value3");
-      AQL_EXECUTE("FOR i IN " + c.name() + " UPDATE i WITH { value2: i.value, value3: i.value } IN " + c.name());
-
-      var query = "FOR i IN " + c.name() + " FILTER (i.value2 == 2 && i.value3 == 2) || (i.value2 == 3 && i.value3 == 3) RETURN i.value2";
+    testUseIndexMultipleFiltersWithLimit : function () {
+      c.insert([{ value: "one" },
+                { value: "two" },
+                { value: "one" },
+                { value: "two" }]);
+      var query = "FOR i IN " + c.name() + " FILTER i.value IN ['one', 'two'] SORT i.value DESC LIMIT 0, 2 FILTER i.value == 'one' RETURN i.value";
 
       var plan = AQL_EXPLAIN(query, {}, opt).plan;
       var nodeTypes = plan.nodes.map(function(node) {
-        if (node.type === "IndexNode") {
-          assertEqual("skiplist", node.indexes[0].type);
-          assertFalse(node.indexes[0].unique);
-          assertEqual([ "value2", "value3" ], node.indexes[0].fields);
-        }
         return node.type;
       });
 
+      assertEqual("SingletonNode", nodeTypes[0], query);
       assertNotEqual(-1, nodeTypes.indexOf("IndexNode"), query);
 
       var results = AQL_EXECUTE(query, {}, opt);
-      assertEqual([ 2, 3 ], results.json.sort(), query);
-      assertEqual(2, results.stats.scannedIndex);
+      assertEqual([ ], results.json, query);
       assertEqual(0, results.stats.scannedFull);
+      assertTrue(results.stats.scannedIndex > 0);
+
+      // retry without index
+      var idx = c.lookupIndex({ type: "skiplist", fields: [ "value" ] });
+      c.dropIndex(idx);
```

Comment:
```suggestion
      c.dropIndex(idx);
      idx = null;
```

Reply:
I may have missed this in a few more places, but you should definitely double-check that you null the variable every time you drop an index, since you use the null check for existence.